### PR TITLE
Make libFuzzer use env variables from fuzzer.options

### PR DESCRIPTION
**Pull request by catenacyber**
@oliverchang this completes https://github.com/google/clusterfuzz/pull/2874

Before that commit, Only AFL can use env variables from fuzzer.options
And I would like to have it for libFuzzer as it is the engine used with golang to get `GODEBUG` set before running my target

### CODE REVIEW

**Comment by catenacyber** [comment]
Happy new year @oliverchang. Could you please review (and merge) this for https://github.com/golang/go/issues/49075 ?

**Review by oliverchang — APPROVED** [review]

**Comment by catenacyber** [comment]
Do you have an error log ?

**Comment by oliverchang** [comment]
See 

```
======================================================================
ERROR: test_analyze_dict (tests.core.bot.fuzzers.libFuzzer.engine_test.ExtraSanitizerIntegrationTests)
Tests recommended dictionary analysis.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspace/src/clusterfuzz/_internal/tests/core/bot/fuzzers/libFuzzer/engine_test.py", line 720, in test_analyze_dict
    options = engine_impl.prepare(corpus_path, target_path, DATA_DIR)
  File "/workspace/src/clusterfuzz/_internal/bot/fuzzers/libFuzzer/engine.py", line 135, in prepare
    for env_var_name, value in extra_env:
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_exit_failure_logged (tests.core.bot.fuzzers.libFuzzer.engine_test.ExtraSanitizerIntegrationTests)
Test that we log when libFuzzer's exit code indicates it ran into an
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspace/src/clusterfuzz/_internal/tests/core/bot/fuzzers/libFuzzer/engine_test.py", line 838, in test_exit_failure_logged
    options = engine_impl.prepare(corpus_path, target_path, DATA_DIR)
  File "/workspace/src/clusterfuzz/_internal/bot/fuzzers/libFuzzer/engine.py", line 135, in prepare
    for env_var_name, value in extra_env:
TypeError: 'NoneType' object is not iterable
```

.... many more repeated like this

File: src/clusterfuzz/_internal/bot/fuzzers/libFuzzer/engine.py [hunk 1]
```diff
@@ -130,6 +131,10 @@ def prepare(self, corpus_dir, target_path, build_dir):
       environment.set_value('USE_EXTRA_SANITIZERS', False)
 
     arguments.extend(strategy_info.arguments)
+    # Update strategy info with environment variables from fuzzer's options.
+    for env_var_name, value in extra_env:
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/bot/fuzzers/libFuzzer/engine.py:135]: You need a None check here, per the test failure.

**Review by oliverchang — CHANGES_REQUESTED** [review]

File: src/clusterfuzz/_internal/bot/fuzzers/libFuzzer/engine.py (hunk 1 above)
Comment (catenacyber) [reply src/clusterfuzz/_internal/bot/fuzzers/libFuzzer/engine.py]: Fair enough, done :-)

**Comment by catenacyber** [comment]
All green now :-)

**Review by oliverchang — APPROVED** [review]

**Comment by catenacyber** [comment]
@oliverchang I do not see my env variable set when running the fuzz target cf https://00f74ba44bc21712537f415f5d5abacc061f1b4587-apidata.googleusercontent.com/download/storage/v1/b/gonids-logs.clusterfuzz-external.appspot.com/o/libFuzzer_gonids_fuzz_parserule%2Flibfuzzer_asan_gonids%2F2023-01-15%2F23:29:36:460564.log?jk=AFshE3Xn8ClzhD4s01x4eyjA-_A6ZgOpjUKX7MMQ6hecqkAeVNr-b90JJeUDs7dbME36uX5-B62PUa5ocXXPlXQXstimGKStBgppZKKPtdpRNWh0OgJhhat5nShNwA3um6ssySr2ZWgV_EGku-rhvgLfC-m3aLlDfVZyawxr4NPHkkPjjUb5DxfhVTph6F7irzLvtzNveiOYeMpKz5kHvhsC3DZKLKdNCpILZNVTVM5suXxdHgHha9K22eSZrlLFC3wkKh0B6HJRgm8w0_tLrLilY0NrhFyA_T4_e7ALU2yQfat59G1ePwS6WkCeo6hP5ElrEKoH1IMjC7Jz7W_Yn9G6cfr27SYxzQRMzsEwGIVkFq7QqqB_Dbpr4p7FXTvIMewIAST79ZUVPo6_h3i0d30s6ndHonPlx-aVHaUtfQyyrXrX_eUr9nJqD_mdj-JIsyz4peg_6SN3RxMxJUsVpIPMLnNP6CYzj6aaWmaEWEXQltROLgxBtqYWEdSUeXmUCc0BZLVOY7C8-YC99SrSMnMeTDV1-JV3aqvAUlU_FgqCJRZnXXBnj-l84bh6tZgY60G72VLWFCgIqxVwjtKK_Z5NR0u4-ixsNEbbIWu8Je2P5f4hcWDDiwr8liELndH0cdso_OH5O34Mz5Q49KEOGUpsfD0F4d9ebkPuXyCCuQNTWmPAdJa09sr29oIw1p8ad1AmtyMeZbJ2NCxV84ffmpQuVCJKChxawM6gxK1CHF0-VE2jTe9MpGqXBJlC44s3j_oOSfWGSYqMQGsPnRisSuvKwjRpvpHcYx63haOQlGlQ4YrfDCjEKG0S93uXtktAKvG9IR-IP8J5C3eOch_GUfhRjg6SZhx7tIOPJ7DPs_yWVS_wLVr6HC74qFlP8VXhxLbCySFHdxjomLJV1l42L7rNcSOfYshG_WKcsR4pfkBlPfBqjRS7lCth2SajVG4CK93LbN_Q9nBDFLMNUQBv3qM4Yy8mplnSz57VBiprTC3ojEvJiB_eOHGEtZzCYd4Z7RI6GMdJ60o4UmFi4PAa5Y8loAXxTyPFkddJRgcSTcw_w3OVmhjX0YNNV58Kku-UF3kllDvhXm02ELcsHGOYo13nBa6cbAyNch5AKr47eyjehsvggO5_I1JKxBSkyy2YhyZbbSSyNxSYcAv6nniEF-dfCSh5xaqP019K6g2C66tFVymWfw85WH9uX7kgh8Mq5mMg0ASfz1qZif1Debd0XnCY6XPI4G5v2VTrcen1DWBM-s-Vwo6ZddjXKyPIWXKmrs6v7E3Ye7acxEoGX92t7TZR_x2GEQihsW8fIJP7N7T4eA&isca=1 

I do not have `GODEBUG=cgocheck=2` as I expect...

How can I debug this ?

**Pull request merged**